skopt/space.py

```diff
@@ -10,6 +10,16 @@
 from sklearn.utils.fixes import sp_version
 
 
+def encode_none(a):
+    return a if a is not None else '__None__'
+encode_none = np.vectorize(encode_none, otypes='U')
+
+
+def decode_none(a):
+    return a if a != '__None__' else None
+decode_none = np.vectorize(decode_none, otypes='O')
```

Comment:
ah yes :) in the meantime I've done the 'remap anything to numbers' approach you mentioned above, which should be better than this, so I'll push that